src/modules/acbs/acbs.service.test.ts

```diff
@@ -0,0 +1,230 @@
+        })
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const getExternalRatingsPromise = service.getExternalRatingsForParty(partyIdentifier, authToken);
+
+      await expect(getExternalRatingsPromise).rejects.toBeInstanceOf(AcbsResourceNotFoundException);
+      await expect(getExternalRatingsPromise).rejects.toThrow(`Party with identifier ${partyIdentifier} was not found by ACBS.`);
+      await expect(getExternalRatingsPromise).rejects.toHaveProperty('innerError', axiosError);
+    });
+
+    it('throws an AcbsException if ACBS responds with a 400 response that is a string that does not contain "Party not found"', async () => {
+      const axiosError = new AxiosError();
+      axiosError.response = {
+        data: 'some error string',
+        status: 400,
+        statusText: 'Bad Request',
+        headers: undefined,
+        config: undefined,
+      };
+
+      // eslint-disable-next-line jest/unbound-method
+      when(httpService.get)
+        .calledWith(`/Party/${partyIdentifier}/PartyExternalRating`, {
+          baseURL: baseUrl,
+          headers: { Authorization: `Bearer ${authToken}` },
+        })
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const getExternalRatingsPromise = service.getExternalRatingsForParty(partyIdentifier, authToken);
+
+      await expect(getExternalRatingsPromise).rejects.toBeInstanceOf(AcbsException);
+      await expect(getExternalRatingsPromise).rejects.toThrow(`Failed to get the external ratings for the party with id ${partyIdentifier}.`);
+      await expect(getExternalRatingsPromise).rejects.toHaveProperty('innerError', axiosError);
+    });
+
+    it('throws an AcbsException if ACBS responds with a 400 response that is not a string', async () => {
+      const axiosError = new AxiosError();
+      axiosError.response = {
+        data: { errorMessage: valueGenerator.string() },
+        status: 400,
+        statusText: 'Bad Request',
+        headers: undefined,
+        config: undefined,
+      };
+
+      // eslint-disable-next-line jest/unbound-method
+      when(httpService.get)
+        .calledWith(`/Party/${partyIdentifier}/PartyExternalRating`, {
+          baseURL: baseUrl,
+          headers: { Authorization: `Bearer ${authToken}` },
+        })
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const getExternalRatingsPromise = service.getExternalRatingsForParty(partyIdentifier, authToken);
+
+      await expect(getExternalRatingsPromise).rejects.toBeInstanceOf(AcbsException);
+      await expect(getExternalRatingsPromise).rejects.toThrow(`Failed to get the external ratings for the party with id ${partyIdentifier}.`);
+      await expect(getExternalRatingsPromise).rejects.toHaveProperty('innerError', axiosError);
+    });
+
+    it('returns the external ratings for the party if ACBS responds with the external ratings', async () => {
+      const ratingEntityCodeA = valueGenerator.stringOfNumericCharacters();
```

Comment:
Could we maybe extract the value generation for `externalRatingInAcbs` to a separate file? I just find it makes the test a bit hard to read!